### DISCUSSION TITLE
iOS on-device inference for lyrics recognition

### DISCUSSION THREAD

**Discussion by vivienseguy** [openai/whisper#926] [Show and tell]
Hello,
I have been developing the Chord ai app for automatic music transcription for a few years. When whisper came out I didn't think for a second that it could do well on recognising lyrics but when I finally tried it was working pretty well. 
So it took me ~1 month to convert PyTorch checkpoints and rewrite all the python decoding logic in C++, and now you can use the Chord ai to perform on-device lyrics recognition of any song. I have shipped the base model and it's pretty fast on a recent iPhone / iPad. I was thinking of shipping the "small" one but the increase in accuracy was barely noticeable so I decided it was not worth increasing the inference time.
Link to the iOS app to try lyrics recognition (free) https://apps.apple.com/app/id1446177109

Vivien

**Comment by jj09** [comment]
Can you share how you integrated model in iOS app?
